snapcraft/tests/unit/project_loader/grammar_processing/test_part_grammar_processor.py

```diff
@@ -29,7 +29,8 @@
 
 
 def load_tests(loader, tests, ignore):
-    tests.addTests(doctest.DocTestSuite(processor))
+    if snapcraft.ProjectOptions().deb_arch == 'amd64':
+        tests.addTests(doctest.DocTestSuite(processor))
```

Comment:
Nice idea actually.

An alternative I was contemplating is to subclass `DocTestSuite` and add a mock there.

Reply:
The `on` statement [does this differently](https://github.com/snapcore/snapcraft/blob/master/snapcraft/internal/project_loader/grammar/_on.py#L46). That example in the processor should probably be doing something similar.

Reply:
I apparently only voiced this in the chat, and not here, so I'll re-iterate my thoughts for completeness:

To my mind mocking the architecture is counter-intuitive in example code like this. You wouldn't use the mock in production but instead want it to behave differently. That's why I think handling this in `load_tests` is more sensible.